google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -1154,4 +1183,15 @@ private CommonObjectRequestParams commonRequestParams(String key) {
         .setEncryptionKeySha256Bytes(ByteString.copyFrom(keySha256.asBytes()))
         .build();
   }
+
+  private FieldMask fieldMaskGenerator(
+      Map<FieldDescriptor, java.lang.Object> fieldDescriptorObjectMap) {
+    return FieldMask.newBuilder()
+        .addAllPaths(
+            fieldDescriptorObjectMap.entrySet().stream()
+                .filter(x -> x.getValue() != null)
+                .map(e -> e.getKey().getName())
+                .collect(Collectors.toList()))
+        .build();
+  }
```

Comment:
Adding this helper is great, since we'll need to do the same thing for each of the update methods.

We can encapsulate a step further though, to simplify the usage of it.

In protobuf, all of our generated classes extends from `com.google.protobuf.Message`, meaning we can change the method signature to take in any `Message` and yield the FieldMask.

```suggestion
  private FieldMask fieldMaskGenerator(Message message) {
    return FieldMask.newBuilder()
        .addAllPaths(
            message.getAllFields().entrySet().stream()
                .filter(x -> x.getValue() != null)
                .map(e -> e.getKey().getName())
                .collect(Collectors.toList()))
        .build();
  }
```

Reply:
thanks! yep agreed this is simpler.